sqlitecluster/SQLite.cpp

```diff
@@ -91,11 +94,17 @@ SQLite::SharedData& SQLite::initializeSharedData(sqlite3* db, const string& file
     }
 }
 
-sqlite3* SQLite::initializeDB(const string& filename, int64_t mmapSizeGB) {
+sqlite3* SQLite::initializeDB(const string& filename, int64_t mmapSizeGB, bool hctree) {
     // Open the DB in read-write mode.
     SINFO((SFileExists(filename) ? "Opening" : "Creating") << " database '" << filename << "'.");
     sqlite3* db;
-    SASSERT(!sqlite3_open_v2(filename.c_str(), &db, SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE | SQLITE_OPEN_NOMUTEX, NULL));
+    string completeFilename = filename;
+    if (hctree) {
+        // Per the docs here: https://sqlite.org/hctree/doc/hctree/doc/hctree/index.html
+        // We only need to specify the full URL when creating new DBs. Existing DBs will be auto-detected as HC-Tree or not.
+        completeFilename = "file://" + completeFilename + "?hctree=1";
+    }
+    SASSERT(!sqlite3_open_v2(completeFilename.c_str(), &db, SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE | SQLITE_OPEN_NOMUTEX | SQLITE_OPEN_URI, NULL));
```

Comment:
It just allows the path to be interpreted as a URI so that `?hctree=1` is seen as a URI parameter instead of a file name. If we used file names with question marks in them, then we'd have to consider this.